src/common/containers/with-repository.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component, PropTypes } from 'react';
+
+import { setGitHubRepository } from '../actions/repository-input';
+
+function withRepository(WrappedComponent) {
+
+  class WithRepository extends Component {
+
+    componentDidMount() {
+      if (!this.props.repository && this.props.fullName) {
+        this.props.dispatch(setGitHubRepository(this.props.fullName));
+      }
+    }
+
+    componentWillReceiveProps(nextProps) {
+      if (this.props.fullName !== nextProps.fullName) {
+        this.props.dispatch(setGitHubRepository(nextProps.fullName));
+      }
+    }
+
+    componentWillUnmount() {
+
+    }
+
+    render() {
+      const { fullName, ...passThroughProps } = this.props;
+      fullName; // just to 'use' this variable
```

Comment:
@katiefenn You mean passed to the wrapped component?

In dev env React will likely complain on receiving prop that is not defined in PropTypes.
In prod probably nothing.

I disabled the check for this line as @earnubs suggested.